factory/shardingFactory.go

```diff
@@ -38,7 +38,12 @@ func CreateShardCoordinator(
 			return nil, "", err
 		}
 		if selfShardId == core.DisabledShardIDAsObserver {
-			selfShardId = uint32(0)
+			pubKeyBytes, err := pubKey.ToByteArray()
```

Comment:
is line 32 correct? if err == sharding.ErrPublicKeyNotFoundInGenesis -> just genesis nodes?

Reply:
If you refer to the fact that a node can belong to a shard even if it is not in genesis (for example a staked node added after genesis), it is ok, because this is the "genesis shard coordinator". After this, the real shard (coordinator) will be calculated inside the `epoch bootstrapper`

Reply:
ok!